deploy/kube-config/influxdb/grafana-image/run.sh

```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+
+HEADER_CONTENT_TYPE="Content-Type: application/json"
+HEADER_ACCEPT="Accept: application/json"
+
+GRAFANA_USER=${GRAFANA_USER:-admin}
+GRAFANA_PASSWD=${GRAFANA_PASSWD:-admin}
+GRAFANA_PORT=${GRAFANA_PORT:-3000}
+
+INFLUXDB_HOST=${INFLUXDB_HOST:-"monitoring-influxdb"}
+INFLUXDB_DATABASE=${INFLUXDB_DATABASE:-k8s}
+INFLUXDB_PASSWORD=${INFLUXDB_PASSWORD:-root}
+INFLUXDB_PORT=${INFLUXDB_PORT:-8086}
+INFLUXDB_USER=${INFLUXDB_USER:-root}
+
+DASHBOARD_LOCATION=${DASHBOARD_LOCATION:-"/dashboards"}
+
+# Allow access to dashboards without having to log in
+export GF_AUTH_ANONYMOUS_ENABLED=true
+export GF_SERVER_HTTP_PORT=${GRAFANA_PORT}
+
+BACKEND_ACCESS_MODE=${BACKEND_ACCESS_MODE:-proxy}
+INFLUXDB_SERVICE_ENDPOINT=${INFLUXDB_SERVICE_ENDPOINT}
+if [ -n "$INFLUXDB_SERVICE_ENDPOINT" ]; then
+  echo "Influxdb endpoint is provided."
+else
+  echo "Discovering influxdb endpoint..."
+  INFLUXDB_SERVICE_ENDPOINT=$(/influxdb_service_discovery)
```

Comment:
Yes, the intention was to go through the InfluxDB service, instead of hitting specific pods. The term "endpoint" is probably causing some confusion here. I'll rename it to "URL" instead.

Lookup order for InfluxDB service URL:
1. If it's provided in the spec via an environment variable, use it.
2. Call into "influxdb_service_discovery" to check if InfluxDB is exposed as an external service.
3. Fall back to "http://monitoring-influxdb".